app/src/main/java/net/squanchy/signin/SignInActivity.java

```diff
@@ -48,6 +67,30 @@ private GoogleApiClient connectToGoogleApis() {
                 .build();
     }
 
+    private void setBottomSheetCallbackOn(View bottomSheet) {
+        BottomSheetBehavior.from(bottomSheet)
+                .setBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+                    @Override
+                    public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                        switch (newState) {
+                            case BottomSheetBehavior.STATE_HIDDEN:
+                                finish();
+                                break;
+                        }
+                    }
+
+                    @Override
+                    public void onSlide(@NonNull View bottomSheet, float slideOffset) {
+                        // no op
+                    }
+                });
+    }
+
+    private void setupWindowParameters() {
```

Comment:
This doesn't really say much. I think this should be called after the result it will produce, since window parameters could be anything. Something along the lines of 'show activity as bottom sheet' or 'update view bounds as bottom sheet' would do.

I don't really have any proper suggestions though so I'm not going to block it 💃